File: app/src/main/java/com/jann_luellmann/thekenapp/data/dao/EventDrinkCrossDAO.java
package com.jann_luellmann.thekenapp.data.dao;

import com.jann_luellmann.thekenapp.data.model.Event;
import com.jann_luellmann.thekenapp.data.model.EventDrinkCrossRef;

import java.util.List;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Transaction;
import androidx.room.Update;

@Dao
public interface EventDrinkCrossDAO {

    @Transaction
    @Query("SELECT * FROM eventDrinkCrossRef")
    LiveData<List<EventDrinkCrossRef>> getAll();

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    void insertAll(EventDrinkCrossRef... eventDrinkCrossRefs);
}

File: app/src/main/java/com/jann_luellmann/thekenapp/data/dao/EventDrinkCrossDAO.kt
package com.jann_luellmann.thekenapp.data.dao

import androidx.lifecycle.LiveData
import androidx.room.*
import com.jann_luellmann.thekenapp.data.model.EventDrinkCrossRef

@Dao
interface EventDrinkCrossDAO {
    @get:Query("SELECT * FROM eventDrinkCrossRef")
    @get:Transaction
    val all: LiveData<List<EventDrinkCrossRef>>

    @Query("SELECT * FROM eventDrinkCrossRef WHERE eventId = :eventId")
    fun findAllById(eventId: Long) : LiveData<List<EventDrinkCrossRef>>

    @Query("SELECT * FROM eventDrinkCrossRef WHERE eventId = :eventId")
    fun findAllByIdBlocking(eventId: Long) : List<EventDrinkCrossRef>

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    fun insertAll(vararg eventDrinkCrossRefs: EventDrinkCrossRef)

    @Delete
    fun delete(eventDrinkCrossRef: EventDrinkCrossRef)
}